src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -83,50 +85,55 @@ export const useTxActions = (): TxActions => {
       return await dispatchTxSigning(safeTx, version, onboard, chainId, txId)
     }
 
-    const signTx: TxActions['signTx'] = async (safeTx, txId, origin) => {
+    const signTx: TxActions['signTx'] = async (safeTx, txId, origin, transaction) => {
       assertTx(safeTx)
       assertWallet(wallet)
       assertOnboard(onboard)
 
+      const humanDescription = transaction?.transaction?.txInfo?.humanDescription
+
       // Smart contract wallets must sign via an on-chain tx
       if (await isSmartContractWallet(wallet)) {
         // If the first signature is a smart contract wallet, we have to propose w/o signatures
         // Otherwise the backend won't pick up the tx
         // The signature will be added once the on-chain signature is indexed
         const id = txId || (await proposeTx(wallet.address, safeTx, txId, origin)).txId
-        await dispatchOnChainSigning(safeTx, id, onboard, chainId)
+        await dispatchOnChainSigning(safeTx, id, onboard, chainId, humanDescription)
         return id
       }
 
       // Otherwise, sign off-chain
-      const signedTx = await dispatchTxSigning(safeTx, version, onboard, chainId, txId)
+      const signedTx = await dispatchTxSigning(safeTx, version, onboard, chainId, txId, humanDescription)
```

Comment:
Should we also pass the whole transaction object here instead of `humanDescription`?

Reply:
We have all the info to build a humanDescription outside of the dispatch functions so I would suggest we keep all the logic there and only pass the string to keep those functions as pure as possible. I would move the fallbacks outside as well.